docs/rules/no-useless-path-segments.md

```diff
@@ -46,3 +50,24 @@ import ".";
 import "..";
 import fs from "fs";
 ```
+
+## Options
+
+### detectUselessIndex
+
+If you want to detect unnecessary `/index` imports in your paths, you can enable the option `detectUselessIndex`. By default it is set to `false`:
+```js
+"import/no-useless-path-segments": ["error", {
+  detectUselessIndex: true,
+}]
+```
+
+Additionally to the patterns described above, the following imports are considered problems if `detectUselessIndex` is enabled:
+
+```js
+// in my-project/app.js
+import "./helpers/index"; // should be "./helpers" (not auto-fixable because of helpers.js)
```

Comment:
this should be autofixable to `./helpers/`

Reply:
Ah. I didn't even know that there is a difference between `./helpers/` --> `./helpers/index.js` and `./helpers` --> `./helpers.js`. 

Just to make sure we don't miss an option, these are all possible import paths:
```
./helpers.js --> ./helpers.js
./helpers --> ./helpers.js
./helpers/ --> ./helpers/index.js
./helpers/index --> ./helpers/index.js
./helpers/index/ --> ./helpers/index/index.js
./helpers/index.js --> ./helpers/index.js
```

If I'm not mistaken, this means that there is no option for ambiguous imports. Either it's an `index.js` file in an directory (in this case, we have a trailing `/`) or it's a file (in this case, we have no trailing `/`). Am I missing something?

Reply:
Yes.

In the presence of both `helpers.js` and `helpers/index.js`:
```sh
./helpers.js → ./helpers.js
./helpers → ./helpers.js (ambiguous)
./helpers/ → ./helpers/index.js
./helpers/index.js → ./helpers/index.js
./helpers/index → ./helpers/index.js
/.helpers/index/ → ./helpers/index/index.js (error, nonexistent here)
```

When there is only `helpers.js`:
```sh
./helpers.js → ./helpers.js
./helpers → ./helpers.js
./helpers/ → ./helpers/index.js (error, nonexistent here)
./helpers/index.js → ./helpers/index.js (error, nonexistent here)
./helpers/index → ./helpers/index.js (error, nonexistent here)
/.helpers/index/ → ./helpers/index/index.js (error, nonexistent here)
```

When there is only `helpers/index.js`:
```sh
./helpers.js → ./helpers.js  (error, nonexistent here)
./helpers → ./helpers/index.js
./helpers/ → ./helpers/index.js
./helpers/index.js → ./helpers/index.js
./helpers/index → ./helpers/index.js
/.helpers/index/ → ./helpers/index/index.js (error, nonexistent here)
```

Reply:
Thank you very much! That helped a lot!

I'll extend the list by the expected result:

### `helpers.js` and `helpers/index.js` exist
Import path (`import '...'`) | Resolves to | Shortest possible path? (Explanation)
-- | -- | --
`./helpers.js` | `./helpers.js` | ✅ (changing the path to `./helpers` would make it ambiguous)
`./helpers` | `./helpers.js` or `./helpers/index.js` (**ambiguous**) | ✅ (path is ambiguous but contains no useless path segments)
`./helpers/` | `./helpers/index.js` | ✅ (changing the path to `./helpers` would make it ambiguous)
`./helpers/index.js` | `./helpers/index.js` | ❌ (change to `./helpers/` possible)
`./helpers/index` | `./helpers/index.js` | ❌ (change to `./helpers/` possible)
`./helpers/index/` | `./helpers/index/index.js` | ~~(Not existent)~~

### Only `helpers.js` exists:
Import path (`import '...'`) | Resolves to | Shortest possible path? (Explanation)
-- | -- | --
`./helpers.js` | `./helpers.js` | ❌ (change to `./helpers` possible)
`./helpers` | `./helpers.js` | ✅ 
`./helpers/` | `./helpers.js` | ❌ (change to `./helpers` possible)
`./helpers/index.js` | `./helpers/index.js` | ~~(Not existent)~~
`./helpers/index` | `./helpers/index.js` | ~~(Not existent)~~
`./helpers/index/` | `./helpers/index/index.js` | ~~(Not existent)~~

### Only `helpers/index.js` exists:
Import path (`import '...'`) | Resolves to | Shortest possible path? (Explanation)
-- | -- | --
`./helpers.js` | `./helpers.js` | ❌ (change to `./helpers` possible)
`./helpers` | `./helpers.js` | ✅ 
`./helpers/` | `./helpers.js` | ❌ (change to `./helpers` possible)
`./helpers/index.js` | `./helpers/index.js` | ❌ (change to `./helpers` possible)
`./helpers/index` | `./helpers/index.js` | ❌ (change to `./helpers` possible)
`./helpers/index/` | `./helpers/index/index.js` | ~~(Not existent)~~

I'll add tests for all these cases and adapt the implementation to that.

Reply:
I've updated your table for two cases, otherwise looks good.